test/services/invoices/delete_invoice.service.test.js

```diff
@@ -289,77 +300,77 @@ describe('Delete Invoice service', () => {
-        const invoice = await InvoiceModel.query().findOne({ billRunId: billRun.id })
-
         await DeleteInvoiceService.go(invoice, billRun.id)
 
-        const licences = await LicenceModel.query().select().where({ billRunId: billRun.id })
+        const licences = await LicenceModel.query().select().where({ invoiceId: invoice.id })
+
         expect(licences).to.be.empty()
       })
 
       it('deletes the invoice transactions', async () => {
-        const invoice = await InvoiceModel.query().findOne({ billRunId: billRun.id })
-
         await DeleteInvoiceService.go(invoice, billRun.id)
 
-        const transactions = await TransactionModel.query().select().where({ billRunId: billRun.id })
+        const transactions = await TransactionModel.query().select().where({ invoiceId: invoice.id })
+
         expect(transactions).to.be.empty()
       })
     })
 
     describe("and it's the only invoice in the bill run", () => {
       beforeEach(async () => {
-        await CreateTransactionService.go(payload, billRun, authorisedSystem, regime)
-        invoice = await InvoiceModel.query().findOne({ billRunId: billRun.id })
+        transaction = await NewTransactionHelper.create()
+        invoice = await InvoiceModel.query().findById(transaction.invoiceId)
+        billRun = await BillRunModel.query().findById(transaction.billRunId)
+
         await billRun.$query().patch({ status: 'NOT_INITIALISED' })
       })
 
       it("changes the bill run status to 'initialised'", async () => {
         await DeleteInvoiceService.go(invoice, billRun.id)
 
-        const result = await BillRunModel.query().findById(billRun.id)
+        billRun = await billRun.$query()
 
-        expect(result.status).to.equal('initialised')
+        expect(billRun.status).to.equal('initialised')
       })
     })
 
     describe('and there are other invoices in the bill run', () => {
       beforeEach(async () => {
-        await CreateTransactionService.go(payload, billRun, authorisedSystem, regime)
-        await CreateTransactionService.go({
-          ...payload,
-          customerReference: 'SOMEONE_ELSE'
-        }, billRun, authorisedSystem, regime)
+        transaction = await NewTransactionHelper.create()
+        invoice = await InvoiceModel.query().findById(transaction.invoiceId)
+        billRun = await BillRunModel.query().findById(transaction.billRunId)
+        await billRun.$query().patch({ status: 'NOT_INITIALISED' })
 
-        invoice = await InvoiceModel.query().findOne({
-          billRunId: billRun.id,
-          customerReference: payload.customerReference
-        })
+        const otherInvoice = await NewInvoiceHelper.create(billRun, { customerReference: 'SOMEONE_ELSE' })
+        const otherLicence = await NewLicenceHelper.create(otherInvoice, { licenceNumber: 'OTHER_LICENCE' })
+        const otherTransaction = await NewTransactionHelper.create(otherLicence, { customerReference: 'SOMEONE_ELSE' })
```

Comment:
See below comment; if we remove that block then we no longer need to assign the transaction to anything as we don't refer to it again.

```suggestion
        await NewTransactionHelper.create(otherLicence, { customerReference: 'SOMEONE_ELSE' })
```